tests/expressions/tests.py

```diff
@@ -534,6 +534,33 @@ def test_subquery_references_joined_table_twice(self):
         outer = Company.objects.filter(pk__in=Subquery(inner.values('pk')))
         self.assertFalse(outer.exists())
 
+    def test_filter_by_count_annotated_inside_subquery(self):
+        Number.objects.create(integer=1000, float=1.2)
+        Employee.objects.bulk_create([
+            Employee(firstname="A", lastname="Z", salary=1000),
+            Employee(firstname="B", lastname="Y", salary=1000),
+            Employee(firstname="C", lastname="X", salary=1000),
+            Employee(firstname="D", lastname="W", salary=1500),
```

Comment:
You can drop all the `firstname`, `lastname` above and default to `''`. Also could create a single `Employee` and use `cnt__gt=0` below.